app/web/src/components/Navigation/Navigation.tsx

```diff
@@ -163,6 +254,65 @@ export default function Navigation() {
     </div>
   );
 
+  const menuItems = menuDropDown(data).map(
+    ({
+      name,
+      notificationCount,
+      route,
+      target,
+      externalLink,
+      hasBottomDivider,
+    }) => {
+      const hasNotification =
+        notificationCount !== undefined && notificationCount > 0;
+
+      const linkContent = (
+        <MenuItem
+          hasNotification={hasNotification}
+          hasBottomDivider={hasBottomDivider}
+        >
+          {hasNotification ? (
+            <Badge color="primary" variant="dot" className={classes.badge}>
+              <Typography noWrap>{name}</Typography>
+            </Badge>
+          ) : (
+            <Typography noWrap>{name}</Typography>
+          )}
+
+          {hasNotification ? (
+            <Typography
+              noWrap
+              variant="subtitle2"
+              className={classes.notificationCount}
+            >
+              {`${notificationCount} unseen`}
+            </Typography>
+          ) : null}
+        </MenuItem>
+      );
+
+      return externalLink ? (
+        <a
+          href={route}
+          key={name}
+          target={target}
+          onClick={() => setMenuOpen(false)}
+        >
+          {linkContent}
+        </a>
+      ) : (
+        <Link
+          to={route}
+          target={target}
+          key={name}
+          onClick={() => setMenuOpen(false)}
+        >
+          {linkContent}
+        </Link>
+      );
```

Comment:
Actually... one more thing 😅  just noticed this `<a>` is wrapping the `MenuItem` which generates invalid HTML... it should be the other way around as `Menu` is rendering as an `ul` under the hood.

i.e. this logic should be within `<MenuItem>`:
```jsx
<MenuItem>
  {
externalLink ? (
        <a
          href={route}
          key={name}
          target={target}
          onClick={() => setMenuOpen(false)}
        >
          {linkContent}
        </a>
      ) : (
        <Link
          to={route}
          target={target}
          key={name}
          onClick={() => setMenuOpen(false)}
        >
          {linkContent}
        </Link>
      );
  }
</MenuItem>
```

etc (excuse the bad formatting, you'll probably fix it easily with Prettier when you save with the editor locally)

Reply:
This might be my bad 😅

Reply:
Why's that? Did you suggest this initially? :p

Reply:
I think so